src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-logic.cpp

```diff
@@ -541,6 +541,40 @@ PushAvStreamTransportServerLogic::HandleAllocatePushTransport(CommandHandler & h
     Commands::AllocatePushTransportResponse::Type response;
     auto & transportOptions = commandData.transportOptions;
 
+    IngestMethodsEnum ingestMethod = commandData.transportOptions.ingestMethod;
+
+    bool isFormatSupported = false;
+
+    for (auto & supportsFormat : mSupportedFormats)
+    {
+        if ((supportsFormat.ingestMethod == ingestMethod) &&
+            (supportsFormat.containerFormat == commandData.transportOptions.containerOptions.containerType))
+        {
+            isFormatSupported = true;
+        }
+    }
+
+    if (isFormatSupported == false)
+    {
+        auto status = to_underlying(StatusCodeEnum::kInvalidCombination);
+        ChipLogError(Zcl,
+                     "HandleAllocatePushTransport[ep=%d]: Invalid Ingest Method and Container Format Combination : (Ingest Method: "
+                     "%02X and Container Format: %02X)",
+                     mEndpointId, to_underlying(ingestMethod),
+                     to_underlying(commandData.transportOptions.containerOptions.containerType));
+        handler.AddClusterSpecificFailure(commandPath, status);
+        return std::nullopt;
+    }
+
+    /*Spec issue for invalid Trigger Type: https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/11701*/
+    if (transportOptions.triggerOptions.triggerType == TransportTriggerTypeEnum::kUnknownEnumValue)
```

Comment:
This check, and other constraint checks (is that what ValidateIncomingTransportOptions is doing?), needs to be before the logic that was added above, per spec.

The fact that the tests are not correctly testing the spec here is bad too.

Reply:
Yeah, all of the constraint validation appears to be in ValidateIncomingTransportOptions (line 578), which should be moved to line 543.